src/Components/GenericExpressCheckout/CustomerRegistrationUtil.php

```diff
@@ -28,57 +29,72 @@ public function getCustomerDataBagFromGetCheckoutSessionResponse(array $response
     {
         $salutationId = $this->getSalutationId($context);
 
+        $billingAddress = array_filter([
+            'salutationId' => $salutationId,
+            'company' => $this->extractBillingData($response, 'company'),
+            'firstName' => $this->extractBillingData($response, 'firstname'),
+            'lastName' => $this->extractBillingData($response, 'lastname'),
+            'street' => $this->extractBillingData($response, 'street'),
+            'additionalAddressLine1' => $this->extractBillingData($response, 'addressaddition'),
+            'zipcode' => $this->extractBillingData($response, 'zip'),
+            'city' => $this->extractBillingData($response, 'city'),
+            'countryId' => $this->getCountryIdByCode($this->extractBillingData($response, 'country') ?? '', $context),
+            'phone' => $this->extractBillingData($response, 'telephonenumber'),
+        ]);
+
+        $shippingAddress = array_filter([
+            'salutationId' => $salutationId,
+            'company' => $this->extractShippingData($response, 'company'),
+            'firstName' => $this->extractShippingData($response, 'firstname'),
+            'lastName' => $this->extractShippingData($response, 'lastname'),
+            'street' => $this->extractShippingData($response, 'street'),
+            'additionalAddressLine1' => $this->extractShippingData($response, 'addressaddition'),
+            'zipcode' => $this->extractShippingData($response, 'zip'),
+            'city' => $this->extractShippingData($response, 'city'),
+            'countryId' => $this->getCountryIdByCode($this->extractShippingData($response, 'country') ?? '', $context),
+            'phone' => $this->extractShippingData($response, 'telephonenumber'),
+        ]);
+
+        $isBillingAddressComplete = $this->hasAddressRequiredData($billingAddress);
+        $isShippingAddressComplete = $this->hasAddressRequiredData($shippingAddress);
+
+        if (!$isBillingAddressComplete && !$isShippingAddressComplete) {
+            throw new RuntimeException($this->translator->trans('PayonePayment.errorMessages.genericError'));
```

Comment:
You did it exactly the same way in the “getSalutationId” function, so that's what I did. Was there a reason why you threw the generic error? Should both be adjusted then?

Reply:
i am not sure what you mean. 
I did this for a missing salutation. Shopware does have the default salutation "not_specified", and if this does not exist, the generic error got thrown. (at this point a lookup in the logfiles is required (and also in code), because the user can not fix this)

there is no other case, where i throw this exception - or did i miss something?

Reply:
I have added an error log.